File: code/14-scripting/example_14.59-67/misc.py
# Python code from Examples 14.59 through 14.65

colors = ["red", "green", "blue"]
complements = {"red" : "cyan",
               "green" : "magenta", "blue" : "yellow"}
print(colors[2], complements["blue"])

crimson = (0xdc, 0x14, 0x3c)    # R,G,B components
print(crimson[1])               # 0x14 = 20

a = 2
b = 3
a, b = b, a
print(a, b)                     # 3 2

X = set(['a', 'b', 'c', 'd'])   # set constructor
Y = set(['c', 'd', 'e', 'f'])   #     takes array as parameter
print(X | Y)                    # (['a', 'b', 'c', 'd', 'e', 'f'])
I = X & Y
print(I)                        # (['c', 'd'])
print(X - Y)                    # (['a', 'b'])
print(X ^ Y)                    # (['a', 'b', 'e', 'f'])
print('c' in I)                 # True

matrix = {}
matrix[2, 3] = 4

File: code/14-scripting/example_14.59-67/misc.rb
# Ruby code from Examples 14.59 through 14.65

a = "4"
# print a + 3, "\n"         # error
print a + 3.to_s, "\n"      # 43
print a.to_i + 3, "\n"      #  7

a = []
a[3] = "1"
print a[0], "\n"                            # blank line
print a[3], "\n"                            # "1"
a[4] = "2"
print a[3] + String(a[0]), "\n"             # "1"
print Integer(a[3]) + Integer(a[4]), "\n"   # 3

colors = ["red", "green", "blue"]
complements = {"red" => "cyan",
               "green" => "magenta", "blue" => "yellow"}
print colors[2], "\n", complements["blue"], "\n"

c1 = colors[2]
print c1, "\n"
c2 = colors.[](2)
print c2, "\n"

colors[2] = "vermilion"
print colors[2], "\n"

colors.[]=(2, "chartreuse")
print colors[2], "\n"
